### PR TITLE
Support for new Microsoft.Data.SqlClient

### PR DESCRIPTION
**Pull request by remunda**
As Microsoft separated SqlClient to new repository and renamed namespaces, eventNames has also changed names. 

The prefix of events in DiagnosticsListener changed to `"Microsoft.Data.SqlClient."`

https://github.com/dotnet/SqlClient/blob/a536885e4e6ccf2a98a86657609b6b72003c890c/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs#L19

### CODE REVIEW

**Comment by remunda** [comment]
@cwe1ss Hi, can you please look at this PR please. Or are there any maintainers? :)

File: src/OpenTracing.Contrib.NetCore/CoreFx/SqlClientDiagnostics.cs
Comment (cwe1ss) [inline src/OpenTracing.Contrib.NetCore/CoreFx/SqlClientDiagnostics.cs:35]: Thank you for the PR! I'm not sure if this will work though, because this currently always casts to `System.Data.SqlClient.SqlCommand` (see `using`-statements). The new `Microsoft.Data.SqlClient`-package introduces new namespaces so any event from that package would pass a `Microsoft.Data.SqlClient.SqlCommand` and therefore result in failing cast. 

I guess we would need either a completely separate class that only handles the `Microsoft.Data.SqlClient`-events (to have a clean separation) or different logic in the `Microsoft.Data.SqlClient.WriteCommandBefore`-case.

If you want this merged, you'd have to look into this please. I'm keeping my efforts on this project to a minimum as OpenTelemetry for .NET has reached RC1 recently and therefore will soon make this project obsolete.

**Review by cwe1ss — CHANGES_REQUESTED** [review]

File: src/OpenTracing.Contrib.NetCore/CoreFx/SqlClientDiagnostics.cs
Comment (cwe1ss) [reply src/OpenTracing.Contrib.NetCore/CoreFx/SqlClientDiagnostics.cs:35]: I just tested this and the casts fail, so I can't merge this. I will however create a different PR with separate diagnostic listeners.

**Pull request closed without merging**
